core/arch/arm/kernel/generic_boot.c

```diff
@@ -551,6 +554,44 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_secure_device_node(void *fdt)
+{
+	const uint32_t *p;
+	int offs;
+	int len;
+#if (TRACE_LEVEL >= TRACE_INFO)
+	char s[33];
+	int n = 0;
+	int m;
+#endif
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "secure-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(secure_device_id))
+		len = sizeof(secure_device_id);
+
+	memcpy(secure_device_id, p, len);
+	secure_device_id_len = len;
+
```

Comment:
There's no path for Normal World to read the DTB from OP-TEE atm.  Indeed we feed it a stub DTB from a-t-f that goes nowhere and has no way to go anywhere else.

Indeed what exists at the minute is a pattern of a-t-f starting both OP-TEE and the Normal World OS, each with their own DTB if they want it, so there is no need for this.  Or do I miss the point?